apps/charterafrica/src/components/Datasets/DatasetCard.js

```diff
@@ -0,0 +1,137 @@
+import formatDateTime from "@/charterafrica/utils/formatDate";
+
+function DatasetCard({
+  formats,
+  notes,
+  title,
+  created,
+  readMore,
+  readLess,
+  updated,
+  author,
+  updatedLabel,
+  createdLabel,
+  sx,
+}) {
+  const [showAll, setShowAll] = useState(false);
+
+  const handleClick = () => setShowAll(!showAll);
+
+  return (
+    <Card
+      sx={{
+        border: "1px solid",
+        borderColor: neutral[50],
+        borderRadius: 0,
+        boxShadow: "none",
+        ...sx,
+      }}
+    >
+      <CardActionArea
+        sx={{
+          ".MuiCardActionArea-focusHighlight": {
+            background: "transparent",
+          },
+        }}
+        component={Link}
+        href="#"
+      >
+        <CardContent>
+          <Grid
+            container
+            spacing={2}
+            justifyContent="space-between"
+            direction={{ xs: "column", md: "row" }}
+          >
+            <Grid item xs={12} md={4}>
+              <LineClampedRichTypography
+                color="neutral.dark"
+                lineClamp={1}
+                order={1}
+                variant="h5SmallSemiBold"
+                sx={{ mb: 1 }}
+              >
+                {title}
+              </LineClampedRichTypography>
+              <Typography variant="p1" color="neutral.main" sx={{ mb: 1 }}>
+                {updatedLabel} {formatDateTime(updated, { includeTime: false })}{" "}
+                | {createdLabel}{" "}
+                {formatDateTime(created, { includeTime: false })}
+              </Typography>
+              <Typography variant="p1SemiBold" color="neutral.main">
+                {author}
+              </Typography>
+            </Grid>
+            <Grid item xs={12} md={4} order={{ xs: 4, sm: 3 }}>
+              <LineClampedRichTypography
+                color="common.black"
+                lineClamp={showAll ? -1 : 3}
+                order={1}
+                variant="p1"
+                sx={{ mb: 1 }}
+              >
+                {notes}
+              </LineClampedRichTypography>
+              <Button
+                sx={{
+                  "&:hover": { backgroundColor: "transparent" },
+                  color: "neutral.dark",
+                  typography: "p1",
+                  textDecoration: "underline",
+                  padding: 0,
+                  minWidth: 0,
+                }}
+                onClick={handleClick}
+              >
+                {showAll ? readLess : readMore}
+              </Button>
+            </Grid>
+            {formats.length ? (
+              <Grid
+                item
+                xs={12}
+                md={4}
+                order={{ xs: 3, sm: 4 }}
+                container
+                justifyContent={{ xs: "flex-start", sm: "flex-end" }}
+                alignItems="center"
+              >
+                {formats?.map((format) => (
```

Comment:
```suggestion
                {formats.map((format) => (
```
You've already checked `formats.length`